vllm/worker/embedding_model_runner.py

```diff
@@ -107,28 +108,52 @@ def execute_model(
             for _ in range(num_layers)
         ]
 
-        execute_model_kwargs = {
-            "input_ids":
-            model_input.input_tokens,
-            "positions":
-            model_input.input_positions,
-            "kv_caches":
-            kv_caches,
-            "attn_metadata":
-            model_input.attn_metadata,
-            **MultiModalInputs.as_kwargs(model_input.multi_modal_kwargs or {},
-                                         device=self.device),
-        }
+        multi_modal_kwargs = model_input.multi_modal_kwargs or {}
+        if (self.observability_config is not None
+                and self.observability_config.collect_model_forward_time):
+            model_forward_start = torch.cuda.Event(enable_timing=True)
+            model_forward_end = torch.cuda.Event(enable_timing=True)
+            model_forward_start.record()
```

Comment:
Copied from GPUModelRunner code. I don't really like how we're repeating this code though...